src/Services/Catalog/Catalog.API/Controllers/CatalogController.cs

```diff
@@ -112,20 +93,12 @@ public async Task<IActionResult> GetItemById(int id)
         public async Task<IActionResult> Items(string name, [FromQuery]int pageSize = 10, [FromQuery]int pageIndex = 0)
         {
 
-            var totalItems = await _catalogContext.CatalogItems
-                .Where(c => c.Name.StartsWith(name))
-                .LongCountAsync();
-
-            var itemsOnPage = await _catalogContext.CatalogItems
-                .Where(c => c.Name.StartsWith(name))
-                .Skip(pageSize * pageIndex)
-                .Take(pageSize)
-                .ToListAsync();
+            var cacheFilter = new CacheCatalogFilter { Key = $"{Request.Path.Value}{Request.QueryString}", PageSize = pageSize, PageIndex = pageIndex };
```

Comment:
same concern as line 45 and if we are doing the same thing in multiple places we should unify it into one function.